src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -193,11 +213,21 @@ public class SetI2CRegister : Cmdlet
 
         protected override void ProcessRecord()
         {
-            this.Device.device.WriteAddressByte(this.Register, this.Data[0]);
-            if (this.PassThru)
+            try
+            {
+                this.Device.device.WriteAddressByte(this.Register, this.Data[0]);
+                if (this.PassThru)
+                {
+                    I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register, this.Data);
+                    WriteObject(result);
+                }
+            }
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
             {
-                I2CDeviceRegisterData result = new I2CDeviceRegisterData(this.Device, this.Register, this.Data);
-                WriteObject(result);
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
+                {
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
+                }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?

---

Updated